app/src/main/java/me/hufman/androidautoidrive/music/controllers/SpotifyAppController.kt

```diff
@@ -219,29 +215,42 @@ class SpotifyAppController(context: Context, val remote: SpotifyAppRemote): Musi
 			if (queueUri != uri) {
 				queueUri = uri
 				queueItems = emptyList()
-				if (uri != null) {
-					val listItem = ListItem(uri, uri, null, playerContext.title, playerContext.subtitle, false, true)
-					loadPaginatedItems(listItem, { queueUri == playerContext.uri }) {
-						queueItems = it
-
-						// shuffle play button somehow gets returned with the rest of the tracks when loading an album
-						if (queueItems.isNotEmpty() && queueItems[0].artist == "" && queueItems[0].title == "Shuffle Play") {
-							queueItems = queueItems.drop(1)
-						}
-
-						// build a basic QueueMetadata while waiting for cover art to load
-						queueMetadata = QueueMetadata(playerContext.title, playerContext.subtitle, queueItems)
-						loadQueueCoverart()
 
-						callback?.invoke(this)
+				val isLikedSongsPlaylist = playerContext.type == "your_library" || playerContext.type == "your_library_tracks"
+				if (isLikedSongsPlaylist) {
+					createLikedSongsQueueMetadata()
+				} else {
+					if (createQueueMetadataJob?.isActive == true) {
+						createQueueMetadataJob?.cancel()
 					}
+					createQueueMetadataJob = null
+					webApi.clearGetLikedSongsAttemptedFlag()
+					createQueueMetadata(playerContext)
 				}
 			}
 
 			callback?.invoke(this)
 		}
 	}
 
+	/**
+	 * Creates the [QueueMetadata] for the "Liked Songs" playlist using the Web API. If the web API is
```

Comment:
I've seen missing cover art on other playlists, such as an Artist playlist from the Library. Can this Web API playlist method be genericized to handle any such playlist with missing coverart?

Reply:
Yeah the Artist playlist is somewhat of another special case as that one not only is missing the cover art for the tracks but also isn't pulling the expected top tracks by the artist like how the Spotify app (desktop and mobile) works. The artist implementation will require some more work to get right and might be a little different so outside of some initial planning I haven't thought about generalizing this just yet as to avoid scope creep and due to the unknowns so far. From my current understanding I think this will be able to be used for other playlists that are having that issue, with some generalizing of course, but I figured I'd handle that when I started working on the artist playlists 😅